src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/Grammar/GrammarGenerator.vb

```diff
@@ -87,7 +87,9 @@ Friend Class GrammarGenerator
                         Dim mappedChildren = structureNode.Children.Select(
                         Function(c)
                             If correspondingChildren.Contains(c) Then
-                                Return c.WithChildKind(correspondingChildrenKinds(c)(local_i))
+                                Return c.WithChildKind(
+                                    If(c.ChildKind(nodeKind.Name),
+                                       correspondingChildrenKinds(c)(local_i)))
```

Comment:
VB grammar has some sections where the token/node relation is specified just by enumerating values in the same order.  however, in some sections the relation is specified by name.  i.e. https://github.com/dotnet/roslyn/blob/a4efc4a972b1ef6615506cf6a0f2a575c6e2f257/src/Compilers/VisualBasic/Portable/Syntax/Syntax.xml#L5863-L5865.  This now handles both kinds.